ofrak_core/ofrak/component/abstract.py

```diff
@@ -149,6 +149,15 @@ async def run(
                     f"the resource context"
                 )
 
+        # Get resources modified by data patches
+        data_ids_to_models = await dependency_handler.map_data_ids_to_resources(
+            patch_result.data_id for patch_result in patch_results
+        )
+        for m in data_ids_to_models.values():
+            modified_resource_ids.add(m.id)
+
+        modified_resource_ids.difference_update(component_context.resources_deleted)
+
```

Comment:
```suggestion
        # Get resourced modified by data patches and add them to 
        # modified_resource_ids
        data_ids_to_models = await dependency_handler.map_data_ids_to_resources(
            patch_result.data_id for patch_result in patch_results
        )
        for m in data_ids_to_models.values():
            modified_resource_ids.add(m.id)
        # Filter deleted resources out of modified_resource_ids
        modified_resource_ids.difference_update(component_context.resources_deleted)
```
If I understand, we are:

1. Adding ids from patch results to `modified_resource_ids`
2. Filtering deleted resource from the modified_resource_ids list.

I think comments could make this clearer,  something like the suggestions. I also don't understand why the patches are not in the modified list already at this point?

Reply:
> I also don't understand why the patches are not in the modified list already at this point?

See my other comment on the nature of the change. The effects of data patches were only used to invalidate attribute dependencies, which *could* result in the resource being "modified", but not *necessarily*.